src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -148,9 +148,143 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
+            .SelectMany(static x => x.Parameters)
+            .Where(static x => x.IsOfKind(CodeParameterKind.RequestConfiguration))
+            .Select(static x => x?.Type?.Name)
+            .Where(static x => !string.IsNullOrEmpty(x))
+            .OfType<string>()
+            .ToList();
+
+        List<CodeInterface> configClasses = codeNamespace.FindChildrenByName<CodeInterface>(elementNames, false)
+            .Where(static x => x != null)
+            .OfType<CodeInterface>()
+            .ToList();
+
+        List<string> queryParamClassNames = configClasses.Where(x => x != null)
+            .Select(static w => w.GetPropertyOfKind(CodePropertyKind.QueryParameters)?.Type?.Name)
+            .Where(static x => !string.IsNullOrEmpty(x))
+            .OfType<string>()
+            .ToList();
+
+        List<CodeInterface> queryParamClasses = codeNamespace.FindChildrenByName<CodeInterface>(queryParamClassNames, false)
+            .Where(static x => x != null)
+            .OfType<CodeInterface>()
+            .ToList();
+
+        List<CodeElement> elements = new List<CodeElement> { codeClass };
+        elements.AddRange(queryParamClasses);
+        elements.AddRange(configClasses);
+
+        codeNamespace.TryAddCodeFile(codeClass.Name, elements.ToArray());
+    }
+
+    private static void CorrectCodeFileUsing(CodeElement currentElement)
+    {
+        // if element is a code file eliminate using references to the same file
+        if (currentElement is CodeFile codeFile && codeFile.Parent is CodeNamespace codeNamespace)
+        {
+            // correct the using values
+            // eliminate the using refering the elements in the same file
+
+            HashSet<string> elementSet = codeFile.GetChildElements(true).Select(x => x.Name.ToLowerInvariant()).ToHashSet();
+            foreach (var element in codeFile.GetChildElements(true))
+            {
+                var startBlockUsings = element switch
+                {
+                    CodeFunction f => f.StartBlock.Usings,
+                    CodeInterface ci => ci.Usings,
+                    CodeEnum ce => ce.Usings,
+                    CodeClass cc => cc.Usings,
+                    _ => Enumerable.Empty<CodeUsing>()
+                };
+
+                var foundUsings = startBlockUsings
+                    .Where(static x => x.Declaration != null && x.Declaration.TypeDefinition != null)
+                    .Where(y => y.Declaration!.TypeDefinition!.GetImmediateParentOfType<CodeNamespace>() == codeNamespace)
+                    .Where(y => elementSet.Contains(y.Declaration!.TypeDefinition!.Name.ToLowerInvariant()));
+
+                foreach (var x in foundUsings)
+                {
+                    switch (element)
+                    {
+                        case CodeFunction ci:
+                            ci.RemoveUsingsByDeclarationName(x.Declaration?.Name!);
+                            break;
```

Comment:
```suggestion
                        case CodeFunction ci when x.Declaration is { } codeType:
                            ci.RemoveUsingsByDeclarationName(codeType.Name);
                            break;
```
If the declaration is indeed nullable, we'll be passing a null value and end up with NullReferenceExcepetions. I think it would be to check and pass a value rather than supressing in this scenario. Same for the other cases in the cases.

Reply:
At this point of execution, the collection has already been filtered and null values excluded